Framework/MDAlgorithms/src/LoadMD.cpp

```diff
@@ -236,11 +237,25 @@ void LoadMD::exec() {
                         << m_QConvention << '\n';
 
     if (pref_QConvention != m_QConvention) {
+      std::vector<double> scaling(m_numDims);
+      for (size_t d = 0; d < m_numDims; d++) {
+        std::string dimd = ws->getDimension(d)->getName();
+        std::string dim2 = dimd.substr(0, 2);
+        dimd = dimd.substr(0, 1);
+
+        // Assume the Q dimensions are those that have names starting with [
+        // such as [H,0.5H,0], or Q_ such as Q_sample_x.
+        // The change in sign should apply only to those.
+        if (dim2 == "Q_" || dimd == "[")
```

Comment:
Would a regex be a better option? One can make stricter checks and more easily add additional patterns in the future. The current solution may be "good enough" and something that can, if necessary, be changed later.